src/m365/teams/commands/chat/chat-member-add.spec.ts

```diff
@@ -0,0 +1,181 @@
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+import { telemetry } from '../../../../telemetry';
+import auth from '../../../../Auth';
+import { Cli } from '../../../../cli/Cli';
+import { CommandInfo } from '../../../../cli/CommandInfo';
+import { Logger } from '../../../../cli/Logger';
+import Command from '../../../../Command';
+import request from '../../../../request';
+import { pid } from '../../../../utils/pid';
+import { session } from '../../../../utils/session';
+import { sinonUtil } from '../../../../utils/sinonUtil';
+import commands from '../../commands';
+const command: Command = require('./chat-member-add');
+
+describe(commands.CHAT_MEMBER_ADD, () => {
+  const chatId = '19:09fd7575940146d383a4a83fc9598546@thread.v2';
+  const userPrincipalName = 'john@contoso.com';
+  const userId = 'a857e888-b602-4790-86d9-3dca2109449e';
+
+  let log: string[];
+  let logger: Logger;
+  let commandInfo: CommandInfo;
+
+  before(() => {
+    sinon.stub(auth, 'restoreAuth').callsFake(() => Promise.resolve());
+    sinon.stub(telemetry, 'trackEvent').callsFake(() => { });
+    sinon.stub(pid, 'getProcessName').callsFake(() => '');
+    sinon.stub(session, 'getId').callsFake(() => '');
+    auth.service.connected = true;
+    commandInfo = Cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.CHAT_MEMBER_ADD);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('adds the member by specifying the userId', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/chats/${chatId}/members`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.doesNotReject(command.action(logger, { options: { chatId: chatId, userId: userId, role: 'guest', verbose: true } }));
```

Comment:
Right now we check nothing. It's perfectly possible that I comment out the entire request in the command and this test will still succeed because the command doesn't throw an exception.
Let's perform a check that the request actually fired, with ideally a check on the request body.